README.md

```diff
@@ -67,46 +67,103 @@ What is needed?
 An object is identified as a built-in using:
 
 * A new `[[Builtin]]` internal slot to mark built-ins
-* A new `@@builtin` symbol (`Symbol.builtin`) property whose default behavior
-  is to provide the value of the `[[Builtin]]` internal slot.
+* A new `@@builtin` symbol (`Symbol.builtin`) property whose value is a function
+  whose default behavior is to provide the value of the `[[Builtin]]` internal
+  slot.
 
 #### `[[Builtin]]` internal slot
 
-All built-ins, with the exception of the intrinsic object `%ObjectPrototype%`,
-would have a `[[Builtin]]` internal slot with a string value identifying the
-name of the built-in. For instance, for the `%Math%` intrinsic object, the
-value of `[[Builtin]]` is `'Math'`.
-
-For `%TypedArray%` intrinsic objects, the value of `[[Builtin]]` is equal to
-the value of `[[TypedArrayName]]`. For instance, for a `%TypedArray%` with
-`[[TypedArrayName]]` `'Uint8Array'`, the value of `[[Builtin]]` is
-`'Uint8Array'`.
+Intrinsic objects listed in the table below have a `[[Builtin]]` internal slot
+with the given string value. Intrinsic objects not listed in the table do *not*
+have the `[[Builtin]]` internal slot.
+
+| Intrinsic Name        | Builtin Name          |
+| --------------------- | --------------------- |
+| `%Array%`             | `'Array'`             |
+| `%ArrayBuffer%`       | `'ArrayBuffer'`       |
+| `%AsyncFunction%`     | `'AsyncFunction'`     |
+| `%Atomics%`           | `'Atomics'`           |
+| `%Boolean%`           | `'Boolean'`           |
+| `%DataView%`          | `'DataView'`          |
+| `%Date%`              | `'Date'`              |
+| `%Error%`             | `'Error'`             |
+| `%EvalError%`         | `'EvalError'`         |
+| `%Float32Array%`      | `'Float32Array'`      |
+| `%Float64Array%`      | `'Float64Array'`      |
+| `%Function%`          | `'function'`          |
+| `%GeneratorFunction%` | `'GeneratorFunction'` |
+| `%Int8Array%`         | `'Int8Array'`         |
+| `%Int16Array%`        | `'Int16Array'`        |
+| `%Int32Array%`        | `'Int32Array'`        |
+| `%JSON%`              | `'JSON'`              |
+| `%Map%`               | `'Map'`               |
+| `%Math%`              | `'Math'`              |
+| `%Number%`            | `'Number'`            |
+| `%Object%`            | `'object'`            |
```

Comment:
The `Builtin.typeof()` function looks at the `constructor`. Since `Object.constructor` is `Function`, calling `Builtin.typeof(Object)` will return `function`. The value here is what is returned for something like `Builtin.typeof({})`